plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/StandardDeployJob.java

```diff
@@ -245,14 +243,34 @@ private String getErrorMessageOrDefault(String defaultMessage) {
     }
   }
 
-  private String getDeployedAppVersion() throws VersionNotFoundException {
+  private AppEngineDeployOutput getDeployOutput() throws CoreException {
     try {
       String rawDeployOutput = deployStdoutLineListener.toString();
-      AppEngineDeployOutput deployOutput = AppEngineDeployOutput.parse(rawDeployOutput);
-      return deployOutput.getVersion();
+      return AppEngineDeployOutput.parse(rawDeployOutput);
     } catch (IndexOutOfBoundsException | JsonParseException ex)  {
-      throw new VersionNotFoundException("Error getting deployed app version", ex);
+      throw new CoreException(StatusUtil.error(getClass(), "Error parsing deploy output", ex));
+    }
+  }
+
+  @VisibleForTesting
+  public String getDeployedAppUrl(AppEngineDeployOutput deployOutput) {
+    String project = deployConfiguration.getProject();
+    Boolean promoting = deployConfiguration.getPromote();
+    String version = deployOutput.getVersion();
+    String service = deployOutput.getService();
+    Boolean usingDefaultService = DEFAULT_SERVICE.equals(service);
+    String appLocation;
```

Comment:
This variable feels superfluous. It looks like you can return the values directly from the blocks.